napari/settings/_base.py

```diff
@@ -516,3 +537,22 @@ def _remove_empty_dicts(dct: dict, recurse=True) -> dict:
         if v == {}:
             del dct[k]
     return dct
+
+
+def _previous_version_subdirectory(root: Path) -> Optional[Path]:
+    napari_dir = root.parent.parent
```

Comment:
Sorry, looking at this now we should change the name of the input variable since we expect it to be the config directory:

```suggestion
def _previous_version_subdirectory(config_dir: Path) -> Optional[Path]:
    napari_dir = config.parent.parent
```

Reply:
Also good to add a short docstring even though this function is private because it's to exactly describe what it does by just it's name. Something like the following.

Returns the subdirectory with a name that corresponds to the most recent version of napari that is older than the current version. Or None if there is no such subdirectory.